client/components/calypso-i18n-provider/index.tsx

```diff
@@ -0,0 +1,31 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect } from 'react';
+import { setLocaleData as setWpI18nLocaleData } from '@wordpress/i18n';
+import { I18nProvider } from '@automattic/react-i18n';
+import i18n from 'i18n-calypso';
+
+const CalypsoI18nProvider: React.FunctionComponent = ( { children } ) => {
+	const [ localeData, setLocaleData ] = React.useState( i18n.state.locale );
+
+	useEffect( () => {
+		const onChange = () => {
+			const nextLocaleData = i18n.state.locale;
```

Comment:
We're using [addTranslations](https://github.com/Automattic/wp-calypso/blob/add/calypso-i18n-component/packages/i18n-calypso/src/i18n.js#L330) to load additional strings from translation chunks, which basically assigns new properties to the original `this.state.locale`, and since [I18nProvider's context value computation is memoized and depends on the passed `localeData`](https://github.com/Automattic/wp-calypso/blob/add/calypso-i18n-component/packages/react-i18n/src/index.tsx#L23), it won't re-compute as it's the same object reference.

As a workaround, you can create a shallow copy `this.state.locale`, e.g. `const nextLocaleData = Object.assign({}, i18n.state.locale);`